projects/centre_console/inc/centre_console_events.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+typedef enum {
+  CENTRE_CONSOLE_EVENT_CAN_RX = 0,
+  CENTRE_CONSOLE_EVENT_CAN_TX,
+  CENTRE_CONSOLE_EVENT_CAN_FAULT,
+  NUM_CENTRE_CONSOLE_EVENTS  // 3
+} CentreConsoleEvent;
+
+typedef enum {
+  CENTRE_CONSOLE_POWER_EVENT_OFF = NUM_CENTRE_CONSOLE_EVENTS,  // 3
```

Comment:
For power distro in the old car, they did NUM+1 instead. I think I agree with that more since then there's no overlap between the definitions. Also since you have more than two sets of events ... I think it might be better to disclude the "NUM_*_EVENTS" part since it isn't actually the number of events anymore. Not sure if there's a better way to do it ...

Reply:
yeah, Idk either... and yeah I never really figured out why they did NUM + 1

Reply:
The important thing is that you don't want overlap such that two events have the same ID so it's important that the each event enum has a different value. Not sure what your take on this might be but I think it might be better to do a big enum and then outline the groupings by using variable names and comments ... i.e.
```c
typedef enum {
   // CENTRE_CONSOLE_CAN_EVENT
  CENTRE_CONSOLE_CAN_EVENT_CAN_RX = 0,
  CENTRE_CONSOLE_CAN_EVENT_CAN_TX,
  CENTRE_CONSOLE_CAN_EVENT_CAN_FAULT,

   // CENTRE_CONSOLE_POWER_EVENT
  CENTRE_CONSOLE_POWER_EVENT_OFF
  CENTRE_CONSOLE_POWER_EVENT_ON_MAIN,
  CENTRE_CONSOLE_POWER_EVENT_ON_AUX,
  CENTRE_CONSOLE_POWER_EVENT_CLEAR_FAULT,
  CENTRE_CONSOLE_POWER_EVENT_PUBLISH_FAULT,
  CENTRE_CONSOLE_POWER_EVENT_FAULT,
  CENTRE_CONSOLE_POWER_EVENT_FAULT_POWER_MAIN_SEQUENCE,
  CENTRE_CONSOLE_POWER_EVENT_FAULT_POWER_AUX_SEQUENCE,

  NUM_CENTRE_CONSOLE_EVENTS
};
```

Reply:
We'd have to prefix everything with CENTRE_CONSOLE, and in the case of DRIVE_FSM_ events it's gonna be just too long and inconvenient. Also, the compiler will sometimes give us warnings if we use the wrong Enum data type when declaration and set it to another enum... having their data types separate will give us that added bonus.

Reply:
That's fair I guess. I don't think we currently take advantage of the typing as far as I can tell.